### PR TITLE
Setup capybara rspec suite

### PR DESCRIPTION
**Pull request by benjaminwood**
This PR shuffles configuration around such that we can share relevant config between the Cucumber suite and the rspec suite. It also removes unneeded or redundant configuration as well as boilerplate comments in helper/configuration files.

Let me know if you have any questions. I've added a couple of example specs that'll be replaced in my next PR with 'real' feature specs.

https://github.com/concord-consortium/rigse/blob/a421a1f9313b4c1809cff77bf8be3e93394eab1d/spec/features/example_feature_spec.rb#L3-L13

I can confirm that headless & non-headless mode works as expected with the second spec which uses chrome to render the page.

### CODE REVIEW

**Comment by benjaminwood** [comment]
Something specific I'd like to draw attention to is this: https://github.com/concord-consortium/rigse/pull/556/files#diff-bbc3aa37353a29906c24d86ee049818aR117

I've removed the symlinks for factories and have used the tools provided by Factory Girl to indicate where the factories should be loaded from. This solved an issue I ran into and also allowed me to remove some duplicate configuration where these factories were being loaded in a similar fashion for data setup.

CI doesn't seem to mind this change and I don't expect it'll have impact on anyone else. Still, it is worth special review.

File: spec/spec_helper_common.rb [hunk 1]
```diff
@@ -134,4 +114,5 @@ def reset(object)
 end
 
 # I don't think this is necessary anymore with the latest factory_girl
```
Comment (scytacki) [inline spec/spec_helper_common.rb]: Can this comment be removed now?

Comment (benjaminwood) [reply spec/spec_helper_common.rb]: Yes, it can. Good catch!

I have removed it.

**Comment by scytacki** [comment]
This test seems to be failing pretty often: https://travis-ci.org/concord-consortium/rigse/jobs/421736560#L1771 it doesn't seem related. Any ideas?

**Comment by benjaminwood** [comment]
Yeah, I was just looking at that. It passes reliably in my dev env. I'm not sure what's up on CI. I've seen quite a few flaky tests in the 'search' job on CI.

My gut tells me it's not related to this PR. From the cucumber suites perspective not a lot has changed. I'd be more suspicious if we had a flaky test in the rspec suite which now has additional configuration via the common helper.

**Comment by benjaminwood** [comment]
Dang, it's still failing upon rebuild.

The good(ish) news is that I just got it to fail locally by running it repeatedly. Unfortunately it only failed once out of ~15 times. So, this will be hard to debug, especially with no backtrace. 

I can dig into it, but I'm not sure it is related to the changes in this PR. It may be related to the shared DB connection and mysql mutex monkeypatches.

**Comment by scytacki** [comment]
OK, I'll merge it.

**Pull request merged**